.github/workflows/smoke-test-release.yml

```diff
@@ -1,185 +1,690 @@
 name: Smoke test release
 on:
+    release:
+        types: [published]
     workflow_dispatch:
         inputs:
-            release_id:
-                description: 'WooCommerce Release Id'
+            tag:
+                description: 'WooCommerce Release Tag'
                 required: true
-
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 permissions: {}
+env:
+    E2E_WP_LATEST_ARTIFACT: e2e-wp-latest--run-${{ github.run_number }}
+    E2E_UPDATE_WC_ARTIFACT: e2e-update-wc--run-${{ github.run_number }}
+    FORCE_COLOR: 1
 
 jobs:
-    login-run:
-        name: Daily smoke test on release.
+    get-tag:
+        name: Get WooCommerce release tag
+        permissions:
+            contents: read
         runs-on: ubuntu-20.04
+        outputs:
+            tag: ${{ steps.tag.outputs.result }}
+            created: ${{ steps.created-at.outputs.created }}
+        steps:
+            - name: Validate tag
+              if: ${{ github.event_name == 'workflow_dispatch' }}
+              env:
+                  GH_TOKEN: ${{ github.token }}
+              run: gh release view "${{ github.event.inputs.tag }}" --repo=woocommerce/woocommerce
+
+            - name: Get tag
+              uses: actions/github-script@v5
+              id: tag
+              with:
+                  result-encoding: string
+                  script: |
+                      console.log( "${{ github.event_name }}" );
+                      return "${{ github.event.release.tag_name }}" || "${{ github.event.inputs.tag }}"
+
+            - name: Verify woocommerce.zip asset
+              env:
+                  GITHUB_TOKEN: ${{ github.token }}
+              run: |
+                  gh release download ${{ steps.tag.outputs.result }} --repo woocommerce/woocommerce
+                  if [[ -f "woocommerce.zip" ]]
+                    then
+                      echo "${{ github.event.inputs.tag }} has a valid woocommerce.zip asset."
+                      exit 0
+                  fi
+
+                  echo "${{ github.event.inputs.tag }} does not have a valid woocommerce.zip asset."
+                  exit 1
```

Comment:
This step verifies if the given release version, whether it came from the `release` or `workflow_dispatch` event, contains a `woocommerce.zip` asset. Fails if it doesn't.

This step is necessary because there are "releases" in our monorepo that do not necessarily belong to WC Core (you'll notice that when you run `gh release list --repo woocommerce/woocommerce`).